cmd/templatizer/app/renderer/render.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package renderer
+
+import (
+	"strconv"
+
+	"github.com/jaegertracing/jaeger/cmd/templatizer/app"
+	esTemplate "github.com/jaegertracing/jaeger/pkg/es"
+	"github.com/jaegertracing/jaeger/plugin/storage/es"
+)
+
+var fixMappingFunc = es.FixMapping
+var loadMappingFunc = es.LoadMapping
+var mapping string
+var err error
+
+// GetMappingAsString returns rendered index templates as string
```

Comment:
I suggest simplifying this to what's below. What's changed:
- Use map of allowed values that acts as a `set` for faster and simpler "contains" check.
- Pass in the `TemplateBuilder` interface to allow mocking. Then just pass `esTemplate.TextTemplateBuilder{}` into this function from `main()`.
- Simplify the logic in this function.

```
var allowedValues = map[string] struct{} {
	"jaeger-span": {},
	"jaeger-service": {},
}

// GetMappingAsString returns rendered index templates as string
func GetMappingAsString(builder esTemplate.TemplateBuilder, opt *app.Options) (string, error) {
	if opt.EsVersion == 7 {
		enableILM, err := strconv.ParseBool(opt.UseILM)
		if err != nil {
			return "", err
		}
		return es.FixMapping(builder, es.LoadMapping("/"+opt.Mapping+"-7.json"), opt.Shards, opt.Replicas, opt.EsPrefix, enableILM)
	}
	return es.FixMapping(builder, es.LoadMapping("/"+opt.Mapping+".json"), opt.Shards, opt.Replicas, "", false)
}

// IsValidOption checks if passed option is a valid index template.
func IsValidOption(val string) bool {
	_, ok := allowedValues[val]
	return ok
```

Reply:
Fixed this in new PR